authnz/src/main/java/io/stargate/auth/SourceAPI.java

```diff
@@ -31,4 +39,27 @@ public enum SourceAPI {
   public String getName() {
     return name;
   }
+
+  public void toCustomPayload(Map<String, ByteBuffer> customPayload) {
+    if (null == customPayload) {
+      return;
+    }
+
+    ByteBuffer buffer = StandardCharsets.UTF_8.encode(this.getName()).asReadOnlyBuffer();
+    customPayload.put(CUSTOM_PAYLOAD_KEY, buffer);
+  }
+
+  public static SourceAPI fromCustomPayload(Map<String, ByteBuffer> customPayload, SourceAPI defaultIfMissing) {
+    if (null == customPayload) {
+      return defaultIfMissing;
+    }
+
+    return Optional.ofNullable(customPayload.get(CUSTOM_PAYLOAD_KEY))
+            .flatMap(buffer -> {
+              String decode = StandardCharsets.UTF_8.decode(buffer).toString();
+              return Arrays.stream(SourceAPI.values()).filter(s -> Objects.equals(s.getName(), decode)).findFirst();
```

Comment:
Why not just use `SourceAPI.valueOf(decode)`? To avoid exception for mismatching name?

Reply:
Because source api has the field `name` that I am comparing to, would not work..

Reply:
Ah. True.